### PR TITLE
DBT-1334: Feat/payments module test

### PR DESCRIPTION
**Pull request by Mrtn-Tianlu**
### Summary of Changes

- Wrote tests for the payments module
- Altered the payments module to pass the tests

Based on #30 
### Testing

``` bash
npm run test modules/subscriptions/subscriptions/subscriptions.test.tsx
```

### CODE REVIEW

File: modules/subscriptions/subscriptions/subscriptions.test.tsx [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+  describe("Subscriptions Page", () => {
+    let Subscriptions: CoreComponent;
+
+    beforeAll(() => {
+      Subscriptions = getComponent("app", "subscriptions");
+    });
+
+    beforeEach(() => {
+      delete window.location;
+      window.location = { href: "" } as Location;
```
Comment (abelbordonado) [inline modules/subscriptions/subscriptions/subscriptions.test.tsx:34]: Tell me more about it
I think in `setup-jest` there is already some mocks for it.

File: modules/subscriptions/subscriptions/subscriptions.test.tsx [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    });
+
+    afterEach(() => {});
+
+    it("Templates is Mounted", async () => {
+      await act(() => render(<Subscriptions />));
+
+      expect(screen.getByText("subscriptions.title")).toBeInTheDocument();
+      expect(screen.getByText("subscriptions.sub-title")).toBeInTheDocument();
+      screen.debug();
```
Comment (abelbordonado) [inline modules/subscriptions/subscriptions/subscriptions.test.tsx]: Delete

File: modules/subscriptions/subscriptions/subscriptions.test.tsx [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    });
+
+    it("Checks subscribe button on second price card redirects", async () => {
+      const mockGetSubscriptionLink = jest
+        .fn<() => Promise<{ url: string }>>()
+        .mockResolvedValue({ url: "https://example.com/subscribe" });
+
+      const serviceGetSpy = jest.spyOn(Service, "get").mockReturnValue({
+        getSubscriptionLink: mockGetSubscriptionLink,
+      } as any);
```
Comment (abelbordonado) [inline modules/subscriptions/subscriptions/subscriptions.test.tsx:99]: You can mock directly the service, as we have the instance `spyOn(subscriptionService, ' getSubscriptionLink`)  .mockOnceResolvedValue({ url: "https://example.com/subscribe" });` (Interesting the use of Once)

That will also remove the `any` here and simplify the block of code

File: modules/subscriptions/subscriptions/subscriptions.test.tsx [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+      });
+    });
+
+    it("Checks table is populated", async () => {
+      render(<Subscriptions />);
+
+      await waitFor(() => {
+        expect(screen.getByTestId("table-row-0"));
+      });
+    });
```
Comment (abelbordonado) [inline modules/subscriptions/subscriptions/subscriptions.test.tsx]: Would merge in one

File: modules/subscriptions/subscriptions/subscriptions.test.tsx [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+      );
+      expect(viewReceiptLink[0]).toBeInTheDocument();
+      expect(viewReceiptLink[0]).toHaveAttribute(
+        "href",
+        expect.stringMatching(/^https:\/\/.+/),
+      );
+      expect(viewReceiptLink[0]).toHaveAttribute("target", "__blank");
+    });
+
+    it.todo("Checks cancel subscription button redirects");
```
Comment (abelbordonado) [inline modules/subscriptions/subscriptions/subscriptions.test.tsx:170]: Duplicated, you did implmeneted just below

File: modules/subscriptions/subscriptions/subscriptions.test.tsx [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+      render(<Subscriptions />);
+
+      const cancelSubBtn = await screen.findByText(
+        "subscription-settings.cancel-sub-btn",
+      );
+      act(() => cancelSubBtn.click());
+
+      expect(cancelSubBtn).toBeInTheDocument();
+    });
+  });
```
Comment (abelbordonado) [inline modules/subscriptions/subscriptions/subscriptions.test.tsx:181]: Test that the date of the subscription is displayed.
Add another test is the same when the status is cancel.
Check what happens when is expired, if we display the page properly (What message error comes)

File: modules/subscriptions/subscriptions/subscriptions.test.tsx (hunk 1 above)
Comment (Mrtn-Omwenga) [reply modules/subscriptions/subscriptions/subscriptions.test.tsx:34]: This is essentially the same thing you were doing in the recording module tests to check when the url changes. Apparently using RouterMock doesn't work when next/router isn't used

File: modules/subscriptions/subscriptions/subscriptions.test.tsx (hunk 5 above)
Comment (Mrtn-Omwenga) [reply modules/subscriptions/subscriptions/subscriptions.test.tsx:170]: It just checks the cancel button exists. I couldn't figure out how to check the redirect. Whatever I tried kept failing

File: modules/subscriptions/subscriptions/subscriptions.test.tsx (hunk 6 above)
Comment (Mrtn-Omwenga) [reply modules/subscriptions/subscriptions/subscriptions.test.tsx:181]: - I can't create a subscription so i can't confirm the sate is today's date
- I couldn't get the cancel subscription to work

Comment (abelbordonado) [reply modules/subscriptions/subscriptions/subscriptions.test.tsx:181]: @Mrtn-Tianlu  We don't create a subscription, we use the mocked service to return a subscription on the api :)
Not sure this answer your question.
The test won't connect with slack

File: modules/subscriptions/subscriptions/subscriptions.test.tsx (hunk 1 above)
Comment (abelbordonado) [reply modules/subscriptions/subscriptions/subscriptions.test.tsx:34]: Yes it wont work. 
My only question is if we can put it on the specific test we want to check (instead on the beforeAll) unless is really needed

File: modules/subscriptions/subscriptions/subscriptions.test.tsx (hunk 5 above)
Comment (abelbordonado) [reply modules/subscriptions/subscriptions/subscriptions.test.tsx:170]: I meant the test is done twice in the code, I think you copy pasted and this one can be deleted
image

**Pull request merged**